src/macro_helpers.h

```diff
@@ -81,43 +81,43 @@ int array[] = { A, B, RESTRICT_ARGS_COUNT(C, 3, B_MACRO, ##__VA_ARGS__) };
-__NN__          that not the lambda but B is what ASSERT_ARGS_COUNT            \
-__NN__          finally evaluates to.                                          \
-__NN__          Please not that passing B through this macro is a must         \
-__NN__          as we need it for the comma operator to work.                  \
-__NN__         */                                                              \
-__NN__   static_assert(sizeof(const char) == sizeof(#__VA_ARGS__ ),            \
-__NN__        /* sizeof(const char) == sizeof(#__VA_ARGS__ ) checks the quoted \
-__NN__           list of additional arguments. If there are none, then the     \
-__NN__           length of #__VA_ARGS__ is a single char as it contains '\0'.  \
-__NN__           This check is not able to find the corner case of a single    \
-__NN__           superfluous comma at the end of the macro arguments as this   \
-__NN__           causes #__VA_ARGS__ being empty (only '\0').                  \
-__NN__         */                                                              \
-__NN__     VERBOSE_STATIC_ASSERT_HEADER                                         \
-__NN__                                                                         \
-__NN__     VERBOSE_FILE_INFO                                                   \
-__NN__     VERBOSE_LINE_INFO                                                   \
-__NL__     "\n"                                                                \
-__NL__     "\nStrange arguments encountered in invocation of " #ORIGINAL_MACRO "."   \
-__NL__     "\n"                                                                \
-__NL__     "\nPlease make sure to pass exactly " #NUM_EXPECTED_ARGS            \
-__NN__                                          " macro arguments to"          \
-__NL__     "\n" #ORIGINAL_MACRO ". Also make sure that there are no dangling"  \
-__NL__     "\ncommas at the end of the argument list."                         \
-__NL__     "\n"                                                                \
-__NL__     "\nThis is the superfluous part at the end of the macro"            \
-__NL__     "\narguments list: \'" #__VA_ARGS__ "\'"                            \
-__NN__                                                                         \
-__NN__     VERBOSE_STATIC_ASSERT_FOOTER                                        \
-__NL__   );                                                                    \
-__NL__                                                                         \
-__NL__   }, /* End of dummy lambda, the comma operator's A operand. */         \
-__NL__   B     /* The overall ASSERT_ARGS_COUNT evaluates to B. */             \
-__NL__   )
+   (                                                                      __NL__ \
+   []{ /* Here we are in the body of a dummy lambda function.             __NL__ \
+          []{} is, BTW, the shortest way to write a lambda.               __NL__ \
```

Comment:
The `__NL__` macro is meant to improve readability of pre-processed code. Adding it to lines that contain only comments will result in a bunch of empty lines in the resulting pre-processed code. I suppose that's unwanted. Because of this, the `__NN__` macro was added. Its sole purpose is not to break the flow of newline macros in the original code and to improve its readability without generating undesired newlines for these lines after pre-processing.

Reply:
Yeah. My thought process was around the tradeoff between the generated code being more compact and the human-editable version of the code being more maintainable. 
Ultimately, I expect that folks will be spending a lot more time with the 'real' implementation than the generated code. And some extra newlines in the generated code feel to me like they're not a significant hardship relative to making editing the code more complicated.

I've actually been wondering if astyle would do a good enough job with the generated code that we could instead just get rid of the __NL__ macros entirely.

Reply:
> And some extra newlines in the generated code feel to me like they're not a significant hardship relative to making editing the code more complicated.

Agreed.

> I've actually been wondering if astyle would do a good enough job with the generated code that we could instead just get rid of the NL macros entirely.

As far as I get it, astyle is currently processing the original code. This won't fix the problems that arise with multi-line pre-processor macros. Every `\` followed by a `\n` is completely removed, and the whole macro ends up in a single, possibly very long line. That's why the `__NL__` is useful.

Of course you could run astyle on the pre-processed code, it will probably add some line breaks. But I prefer the pre-processed code to have the same layout as the original code which aids orientation and code comparison.